lang/java/avro/src/main/java/org/apache/avro/file/DataFileReader.java

```diff
@@ -47,7 +47,7 @@
                                              DatumReader<D> reader)
     throws IOException {
     if (in.length() < MAGIC.length)
-      throw new IOException("Not an Avro data file");
+      throw new EOFException("Not an Avro data file");
```

Comment:
Will this be confusing? Say I want to take files and throw them at the wall to see what sticks. With this change as-is I'll need to see if the IOE is either EOF or IAM. Is that desirable? Do we gain enough by being able to distinguish the case where the magic header was invalid due to not enough bytes?